src/include/rados/librados.hpp

```diff
@@ -1221,7 +1221,7 @@ namespace librados
     int init(const char * const id);
     int init2(const char * const name, const char * const clustername,
 	      uint64_t flags);
-    int init_with_context(config_t cct_);
+    int init_with_context(config_t cct_, const std::string& name = "");
```

Comment:
same here.  it has to be init_with_context_name or something.. don't overload the same name as gonig from non-overloaded to overloaded breaks the C++ ABI